src/matcher/plugin.py

```diff
@@ -336,4 +342,48 @@ def pytest_addoption(parser) -> None:
       , default=False
       )
 
+
+@pytest.hookimpl(trylast=True)
+def pytest_configure(config: pytest.Config):
+    """Register additional configuration."""
+    if config.getoption('--pm-reveal-unused-files'):
+        config.option.collectonly = True
+        reporter = UnusedFilesReporter()
+        config.pluginmanager.unregister(name='terminalreporter')
+        config.pluginmanager.register(reporter, 'terminalreporter')
+
+
+class UnusedFilesReporter:
```

Comment:
Not sure if this class should be "public" in this module… Renaming to `_UnusedFilesReporter` can give a clue to the reader that it's not a public API of the package.

Also, for code grouping purposes I'd prefer to move this class declaration above the `# BEGIN Pytest hooks` block (which contains hooks (function definitions) recognizable by `pytest`).